addons/account/models/account_bank_statement.py

```diff
@@ -170,303 +45,191 @@ def _get_move_line_count(self):
-
+        self.line_ids.is_anchor = False
         for statement in self:
-
-            # Chatter.
-            statement.message_post(body=_('Statement %s confirmed.', statement.name))
-
+            statement.line_ids.sorted()[:1].write({'is_anchor': True,
+                                                   'anchor_value': statement.balance_end_real
+                                                   })
+            # todo: POMA add statement report
             # Bank statement report.
-            if statement.journal_id.type == 'bank':
-                content = self.env["ir.actions.report"]._render_qweb_pdf('account.action_report_account_statement', statement.id)[0]
-                self.env['ir.attachment'].create({
-                    'name': statement.name and _("Bank Statement %s.pdf", statement.name) or _("Bank Statement.pdf"),
-                    'type': 'binary',
-                    'raw': content,
-                    'res_model': statement._name,
-                    'res_id': statement.id
-                })
-
-        self._check_balance_end_real_same_as_computed()
-        self.write({'state': 'confirm', 'date_done': fields.Datetime.now()})
-
-    def button_validate_or_action(self):
-        if self.journal_type == 'cash' and not self.currency_id.is_zero(self.difference):
-            return self.env['ir.actions.act_window']._for_xml_id('account.action_view_account_bnk_stmt_check')
-
-        return self.button_validate()
-
-    def button_reopen(self):
-        ''' Move the bank statements back to the 'open' state. '''
-        if any(statement.state == 'draft' for statement in self):
-            raise UserError(_("Only validated statements can be reset to new."))
-
-        self.write({'state': 'open'})
-        self.line_ids.move_id.button_draft()
-        self.line_ids.button_undo_reconciliation()
-
-    def button_reprocess(self):
-        """Move the bank statements back to the 'posted' state."""
-        if any(statement.state != 'confirm' for statement in self):
-            raise UserError(_("Only Validated statements can be reset to new."))
+            # if statement.journal_id.type == 'bank':
+            #     content = self.env["ir.actions.report"]._render_qweb_pdf(
+            #         'account.action_report_account_statement', statement.id)[:1]
+            #     self.attachment_ids += self.env['ir.attachment'].create({
+            #         'name': statement.name and _("Bank Statement %s.pdf", statement.name) or _("Bank Statement.pdf"),
+            #         'type': 'binary',
+            #         'raw': content,
+            #     })
+
+    def action_post(self):
+        self._post()
+        return {'type': 'ir.actions.act_window_close'}
 
-        self.write({'state': 'posted', 'date_done': False})
+    def action_unpost(self):
+        self.state = 'open'
+        self.line_ids.is_anchor = False
+        self._compute_state()
```

Comment:
all actions should be designed to work on multiple records

Triggering manually a compute is a very bad idea because you are playing outside the ORM bounds...

Reply:
it works on multiple records, non :?
I change the compute part.